modules/pytket-honeywell/pytket/extensions/honeywell/backends/rebase.py

```diff
@@ -0,0 +1,55 @@
+from typing import Set, Union
+from sympy import Expr  # type: ignore
+from pytket.circuit import OpType, Circuit  # type: ignore
+from pytket.passes import SquashCustom, RebaseCustom  # type: ignore
+from pytket.transform import Transform  # type: ignore
+
+
+def _approx_0_mod_2(x: Union[float, Expr], eps: float = 1e-10) -> bool:
+    if isinstance(x, Expr) and not x.is_constant():
+        return False
+    x = float(x)
+    x %= 2
+    return min(x, 2 - x) < eps
+
+
+def _tk1_to_PhasedXRz(
+    a: Union[float, Expr], b: Union[float, Expr], c: Union[float, Expr]
+) -> Circuit:
+    circ = Circuit(1)
+    if _approx_0_mod_2(b - 1):
+        # Angles β ∈ {π, 3π}
+        circ.add_gate(OpType.PhasedX, [b, (a - c) / 2], [0])
+    elif _approx_0_mod_2(b):
+        # Angle β ∈ {0, 2π}
+        circ.Rz(a + b + c, 0)
+    else:
+        circ.Rz(a + c, 0)
+        circ.add_gate(OpType.PhasedX, [b, a], [0])
+
+    Transform.RemoveRedundancies().apply(circ)
+    return circ
+
+
+def _quantinuum_rebase(gate_set: Set[OpType]) -> RebaseCustom:
```

Comment:
Should we check and raise an exception if `gate_set` does not include {Rz, PhasedX, ZZMax}, since the rebase pass won't be correct otherwise? (Then we'd just set `multiops` and `singleops` to these gates below.)

Reply:
Though as implemented that's not actually what's happening, I need to fix that

Reply:
https://github.com/CQCL/pytket-extensions/blob/1271f18fb92404b2677df74c081ad1486252ce44/modules/pytket-qiskit/pytket/extensions/qiskit/backends/aer.py#L150-L156

this is what I actually want